### PR TITLE
Change default ruby version to 3.3.2

### PR DESCRIPTION
**Pull request by shleeable**
Pending upstream images

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,15 +7,15 @@
 ARG TARGETPLATFORM=${TARGETPLATFORM}
 ARG BUILDPLATFORM=${BUILDPLATFORM}
 
-# Ruby image to use for base image, change with [--build-arg RUBY_VERSION="3.3.1"]
-ARG RUBY_VERSION="3.3.1"
+# Ruby image to use for base image, change with [--build-arg RUBY_VERSION="3.3.x"]
```
Comment (mjankowski) [inline Dockerfile:10]: "3.3.x" is an excellent innovation for this file.

**Comment by mjankowski** [comment]
If we rebase this and/or re-trigger CI, I think the version is available now (from other projects)

**Comment by shleeable** [comment]
https://hub.docker.com/_/ruby/tags?page=&page_size=&ordering=&name=3.3.2

Not as of posting... should be up by tomorrow

**Comment by mjankowski** [comment]
Ah, sorry, I meant the CI run portion - https://github.com/ruby/setup-ruby/pull/603 - not the Docker portion.

Noted re: docker hub still waiting.

**Comment by shleeable** [comment]
> https://hub.docker.com/_/ruby/tags?page=&page_size=&ordering=&name=3.3.2
> 
> Not as of posting... should be up by tomorrow

Completed an hour ago... PR is good to go

**Comment by shleeable** [comment]
Tested on A.S without regression.

File: Dockerfile [hunk 2]
```diff
@@ -7,22 +7,22 @@
...
-# Ruby image to use for base image based on combined variables (ex: 3.3.1-slim-bookworm)
+# Ruby image to use for base image based on combined variables (ex: 3.3.x-slim-bookworm)
 FROM docker.io/ruby:${RUBY_VERSION}-slim-${DEBIAN_VERSION} as ruby
 
 # Resulting version string is vX.X.X-MASTODON_VERSION_PRERELEASE+MASTODON_VERSION_METADATA
 # Example: v4.2.0-nightly.2023.11.09+something
 # Overwrite existence of 'alpha.0' in version.rb [--build-arg MASTODON_VERSION_PRERELEASE="nightly.2023.11.09"]
 ARG MASTODON_VERSION_PRERELEASE=""
-# Append build metadata or fork information to version.rb [--build-arg MASTODON_VERSION_METADATA="something"]
+# Append build metadata or fork information to version.rb [--build-arg MASTODON_VERSION_METADATA="PR-12345"]
```
Comment (ClearlyClaire) [inline Dockerfile:25]: Unrelated change, but not a big deal.

**Review by ClearlyClaire — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (shleeable) [reply Dockerfile:25]: I was trying to be more specific than `something`... but I'm happy to revert.

**Review by renchap — APPROVED** [review]

**Comment by vmstan** [comment]
> Tested on A.S without regression.

I can't get it to build anymore.

```
 => ERROR [linux/arm64 build 6/6] RUN   rm /usr/local/bin/yarn*;   corepack enable;   corepack prepare --activate;                                                                                                                              0.2s
------
 > [linux/arm64 build 6/6] RUN   rm /usr/local/bin/yarn*;   corepack enable;   corepack prepare --activate;:
0.131 /usr/bin/env: ‘node’: No such file or directory
------
Dockerfile:148
--------------------
 147 |
 148 | >>> RUN \
 149 | >>> # Configure Corepack
 150 | >>>   rm /usr/local/bin/yarn*; \
 151 | >>>   corepack enable; \
 152 | >>>   corepack prepare --activate;
 153 |
--------------------
ERROR: failed to solve: process "/bin/bash -o pipefail -o errexit -c rm /usr/local/bin/yarn*;   corepack enable;   corepack prepare --activate;" did not complete successfully: exit code: 127
```

**Comment by shleeable** [comment]
Weird. corepack has nothing to do with ruby?

**Comment by vmstan** [comment]
corepack might not but the build process copies node binaries into the image and it's failing for the arm64 builds on 3.3.2 where it's not on amd64.

**Pull request merged**